contrib/vdirsyncer.plist

```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+  <dict>
+    <!-- click requires a proper UTF-8 locale -->
+    <key>EnvironmentVariables</key>
+    <dict>
+           <key>LANG</key>
+           <string>en_US.UTF-8</string>
+           <key>LC_ALL</key>
+           <string>en_US.UTF-8</string>
```

Comment:
I have no problem with including assumptions (even homebrew-specific ones) in the plist because users understand how to change them, but using the wrong locale may break things in a very subtle way on POSIX (corrupted filenames etc), if theoretically your real locale is a very obscure one. I don't know if such a locale is even possible on OS X. But either way, it is technically wrong to assume any hardcoded value.

I figured there must be a way to get the proper locale variables, maybe as an extra command before vdirsyncer. I think at this point we would need some third-party opinion/knowledge.

On 24 April 2016 13:09:59 CEST, benjaminfrank <notifications@github.com> wrote:
>> @@ -0,0 +1,34 @@
>> +<?xml version="1.0" encoding="UTF-8"?>
>> +<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN"
>"http://www.apple.com/DTDs/PropertyList-1.0.dtd">
>> +<plist version="1.0">
>> +  <dict>
>> +    <!-- click requires a proper UTF-8 locale -->
>> +    <key>EnvironmentVariables</key>
>> +    <dict>
>> +           <key>LANG</key>
>> +           <string>en_US.UTF-8</string>
>> +           <key>LC_ALL</key>
>> +           <string>en_US.UTF-8</string>
>
>I don't know what exactly the default is, but vdirsyncer (or rather
>click) bails out with this UTF-8 error (which I can't remember right
>now, but it tells you to properly set your LANG & LC_ALL).
>Launchd does not use you .profile or anything, I wouldn't even know
>which shell they use internally or if the use a proper shell at all or
>rather fork() or something.
>
>In launchd you can set env vars by calling ```launchctl setenv <VAR>
><VALUE>``` however this is not persisted. The only way to "persist"
>such a setting is to create a .plist that sets those values after
>booting. this has (at least) 2 disadvantages: a) execution order can't
>be controlled in launchd b) it creates a dependency on another plist
>
>So I figured setting those vars inside the plist is the most hasslefree
>way of doing things.
>Other plist are usually some startup scripts rather than cron like
>jobs, e.g. in homebrew there is no cron like plist.
>
>The more I think about it, this plist takes quite a few assumptions,
>i.e. where vdirsyncer is installed. Maybe it would make more sense to
>put some comment ala "this is an example, please adjust as needed"o n
>top? For the homebrew one, I need to change this anyway so it properly
>adjust to where homebrew install things etc.
>
>---
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/pimutils/vdirsyncer/pull/435/files/89181d1cb25be28a22d3fe436ab3b5b79c1487b0#r60842852

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.